### PR TITLE
feat(FN-3159): add keying sheet generation with amendments test

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
We don't have amendments on our facilities in our existing keying sheet generation tests.
Rather than adding to an already complex test file covering multiple other edge cases I have chosen to add a new e2e-test specific to this case so it is more readable and maintainable.

## Resolution :heavy_check_mark:
- Add e2e-test for a keying sheet generation journey where the facility in question has amendments to the cover end date

### CODE REVIEW

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/generate-fixed-fee-adjustment-with-amendments.spec.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     * - bank margin: 0.9 (this is a fixed, constant value)
+     * - interest percentage: 100 / 100 = 1
+     * - number of days left in cover period divided by day count basis: 730 / 1 = 730
+     * This yields a current utilisation value of 730 * 0.9 = 657
+     *
+     * The adjustment is therefore the difference between the previous
+     * fixed fee which was 700 and 657 which is a decrease of 43
+     */
+    cy.assertText(pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentDecrease(feeRecordId), '43.00');
+    cy.assertText(pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentIncrease(feeRecordId), '-');
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/generate-fixed-fee-adjustment-with-amendments.spec.js:121]: Would be nice to destructure the `pages` in this test so it reads nicer without losing context e.g

```js
utilisationReportPage.premiumPaymentsTab.generateKeyingDataButton().click();
checkKeyingDataPage.generateKeyingSheetDataButton().click();

cy.assertText(keyingSheetTab.fixedFeeAdjustmentDecrease(feeRecordId), '43.00');
cy.assertText(keyingSheetTab.fixedFeeAdjustmentIncrease(feeRecordId), '-');
```

Not a blocker, seems the majority of tests don't do this

**Review by ttbarnes — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]
